runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -559,18 +559,127 @@ xlpSubOptionsParser(J9JavaVM *vm, IDATA xlpIndex, XlpError *xlpError, UDATA *req
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	IDATA result = j9vmem_find_valid_page_size(0, &pageSize, &pageFlags, &isRequestedSizeSupported);
+
+	/*
+	 * j9vmem_find_valid_page_size happened to be changed to always return 0
+	 * However formally the function type still be IDATA so assert if it returns anything else
+	 */
+	Assert_MM_true(0 == result);
+
+	extensions->requestedPageSize = pageSize;
+	extensions->requestedPageFlags = pageFlags;
+
+	if (!isRequestedSizeSupported) {
+		/* Print a message indicating requested page size is not supported and a different page size will be used */
+		const char *oldQualifier, *newQualifier;
+		const char *oldPageType = NULL;
+		const char *newPageType = NULL;
+		UDATA oldSize = requestedPageSize;
+		UDATA newSize = pageSize;
+		qualifiedSize(&oldSize, &oldQualifier);
+		qualifiedSize(&newSize, &newQualifier);
+		if (0 == (J9PORT_VMEM_PAGE_FLAG_NOT_USED & requestedPageFlags)) {
+			oldPageType = getPageTypeString(requestedPageFlags);
+		}
+		if (0 == (J9PORT_VMEM_PAGE_FLAG_NOT_USED & pageFlags)) {
+			newPageType = getPageTypeString(pageFlags);
+		}
+		if (NULL == oldPageType) {
+			if (NULL == newPageType) {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED, oldSize, oldQualifier, newSize, newQualifier);
+			} else {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED_WITH_NEW_PAGETYPE, oldSize, oldQualifier, newSize, newQualifier, newPageType);
+			}
+		} else {
+			if (NULL == newPageType) {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED_WITH_REQUESTED_PAGETYPE, oldSize, oldQualifier, oldPageType, newSize, newQualifier);
+			} else {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED_WITH_PAGETYPE, oldSize, oldQualifier, oldPageType, newSize, newQualifier, newPageType);
+			}
+		}
+	}
+}
+
+/**
+ * 
+ * Obtains and sets the largest page size available on the system. 
+ * Returns true if the operation was successful.
+ * 
+ */
+bool
+gcSetPreferredPageSize(J9JavaVM *vm, MM_GCExtensions *extensions)
+{
+	UDATA defaultLargePageSize = 0;
+	UDATA defaultLargePageFlags = J9PORT_VMEM_PAGE_FLAG_NOT_USED;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	j9vmem_default_large_page_size_ex(0, &defaultLargePageSize, &defaultLargePageFlags);
+	
+	if (0 != defaultLargePageSize) {
+		extensions->requestedPageSize = defaultLargePageSize;
+		extensions->requestedPageFlags = defaultLargePageFlags;
+	} else {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * 
+ *  Sets the large page size used within the objectheap when the options -XX:+UseLargePages or -XX:LargePageSizeInBytes=<size> are used.
+ *  The value of the large page size will only be overwritten when these arguments are located in the right most index.
+ *  This is compared to the indices of -Xlp, -Xlp<size>, and -Xlp:objectheap:pagesize=<xx>/
+ * 
+ */
+static bool
+gcProcessXXObjectHeapArguments(J9JavaVM *vm)
```

Comment:
Please change name for this function. Currently it pretends all XX Object heap related arguments are handled here. This is not correct. Large Pages size XX arguments are handled here only

Reply:
Thanks for catching that. 
See https://github.com/eclipse/openj9/pull/7476/commits/c406cf97128641d179cafbba2e76aa3c61c16374